node/src/components/consensus/era_supervisor.rs

```diff
@@ -206,21 +206,36 @@ where
         if self.active_eras.contains_key(&era_id) {
             panic!("{} already exists", era_id);
         }
+        let last_era_id = self.current_era;
         self.current_era = era_id;
 
         let sum_stakes: Motes = validator_stakes.iter().map(|(_, stake)| *stake).sum();
         assert!(
             !sum_stakes.value().is_zero(),
             "cannot start era with total weight 0"
         );
+
+        let init_round_exp = self
+            .active_eras
+            .get(&last_era_id)
+            .and_then(|era| {
+                era.consensus
+                    .as_any()
+                    .downcast_ref::<HighwayProtocol<I, HighwayContext>>()
+            })
+            .and_then(|highway_proto| highway_proto.median_round_exp())
+            .unwrap_or(self.highway_config().minimum_round_exponent);
```

Comment:
I'd find it a bit cleaner if in era 0 we didn't even try all that stuff. Maybe something like:
```rust
era_id.checked_sub(1)
    .and_then(|e_id| self.active_eras.get(&e_id))
    .and_then(|era| …) …
```